Add ADR describing `staged-recipes` workflow for recipe testing. (#20)
pangeo-forge/roadmap: doc/adr/0005-administrative-commands-for-pr-testing.md

## Decision

To alleviate this limitation we will include some administrator intervention in the PR review process by using [slash command dispatch](https://github.com/peter-evans/slash-command-dispatch) to allow administrators to comment on a PR in order to register and run a reduced dimension version of the recipe with the development bakery.  So the current workflow will be 

> 1. Recipe author forks `staged-recipes` and develops a new recipe with a `meta.yaml` entypoint.

Reviewer: ```suggestion
1. Recipe author forks `staged-recipes` and develops a new recipe with a `meta.yaml` entrypoint.
```

2. Recipe author submits a PR from their fork to `staged-recipes`.
3. This PR event triggers a workflow with initial linting and code validation that does not require secrets.
4. A `pangeo-forge` admin adds a comment command to register and run the reduced dimension version of the recipe on the development bakery.

> 5. If successful the bakery operator whose bakery id is referenced in `meta.yaml` should review the PR prior to merge into `master`.

Reviewer: On this - Do we currently have a GitHub ID for the bakery operator in the bakery database?

It'd be really slick to get a comment applied to the PR based on that selection, to automatically alert the operator (or we could automatically assign them as a reviewer)

Reviewer 2: Good idea. It would be good to use org-level groups here, rather than a single individual github ID.

Reviewer: @rabernat yeah good call. Well, I guess it's up to the Bakery operator what github handle they put... 

Some folks might not have a whole org setup

Reviewer 2: I was thinking this would need to be a group _inside_ the `pangeo-forge` org. I'm pretty sure we can't make permissions contingent on outside orgs, but I could be wrong.

Reviewer: Ohhh sorry yes. For them to physically approve it, we'd need to give them permissions. We could just do a comment approval 🤷

> 5. If successful the bakery operator whose bakery id is referenced in `meta.yaml` should review the PR prior to merge into `master`.

Reviewer: ```suggestion
5. If successful the bakery operator whose bakery id is referenced in `meta.yaml` should review the PR prior to merge into `main`.
```

Semantics

Author: https://github.com/pangeo-forge/staged-recipes/branches

> 6. The PR is merged into `master` which triggers a workflow to create a new `feedstock` repository for the recipe.

Reviewer: ```suggestion
6. The PR is merged into `main` which triggers a workflow to create a new `feedstock` repository for the recipe.
```

Ditto

Author: https://github.com/pangeo-forge/staged-recipes/branches

Reviewer 2: Do we want to formalize how the new feedstock name will be determined? In https://github.com/pangeo-forge/roadmap/blob/master/doc/adr/0002-use-meta-yaml-to-track-feedstock-metadata.md#top-level-data we define a top-level key called `id` in `meta.yaml`. So maybe
> The new feedstock name will be determined by joining `id` from `meta.yaml` with the string `-feedstock`

There are a few outstanding points here. There has been some [discussion](https://github.com/pangeo-forge/pangeo-forge-recipes/issues/97) about how to expose reduced recipe dimensions but we'll need to finalize an approach for this for repeatable recipe integration testing.  This discussion only outlines the workflows within the `staged-recipes` repo.  There will be an additonal ADR describing the detailed workflows in the `feedstock` repos to follow.

## Consequences

> This will require adminstrator intervention in the PR review process will also add complexity for recipe authors who will need to incorporate reduced testing dimensions into their recipes.

Reviewer: ```suggestion
This will require `pangeo-forge` administrator and bakery operator intervention in the PR review process, it will also add complexity for recipe authors who will need to incorporate reduced testing dimensions into their recipes.
```
